### ISSUE TITLE
Importing Error

### ISSUE THREAD

**Issue by Ningaal** [Zarroc2762/E7-Gear-Optimizer#48]
Hi,

This error appeared after updating to the latest version.

"After parsing a value an unexpected character was encountered: c. Path 'results[0].skills[2].name', line 1, position 20987"

Also

"Corrupted or wrong file format. Please select a JSON file exported by this application!"

My file was exported by this optimizer. In fact, it was automatically uploaded from the last time I have used it before the latest update.

**Comment by Zarroc2762** [comment]
I can't reproduce the issue on my end. A friend of mine had the same problem and fixed it by redownloading the the optimizer and deleting the old files.

**Comment by phootrf** [comment]
![errormessage](https://user-images.githubusercontent.com/63897048/79630160-51067a80-814f-11ea-92bf-a89fa9975b90.PNG)
Unfortunately I have the same issue, does not solve itself by redownloading and overwriting files several times.

**Comment by Zarroc2762** [comment]
I managed to reproduce the issue on my hand. There was a problem with cached files from the old API. Should be fixed in the latest update. Please check if the problem still exists

**Comment by Dalkil** [comment]
Hi! First of all, thanks for this program. 

I've also been having this issue, and using version 2.8.4 didn't solve the issue. 
Also, it seems epicsevendb is not upgrading all new characters. Is there an alternative to it? 

Thanks.

**Comment by phootrf** [comment]
> 
> 
> I managed to reproduce the issue on my hand. There was a problem with cached files from the old API. Should be fixed in the latest update. Please check if the problem still exists

Hello,
I've tried the latest release but unfortunately the problem persists. Also there is now an unhandled directory not found exception. Still, thank you for your work. I'm impressed by the power this tool displays.

![directoryNotFoundException](https://user-images.githubusercontent.com/63897048/79644185-88087a80-81a7-11ea-8ee8-21ea364ddbce.PNG)

**Comment by RaphaelDDL** [comment]
Can you guys try manually clearing the cache as ellesde pointed here https://github.com/Zarroc2762/E7-Gear-Optimizer/issues/43#issuecomment-615913983 and see if works?

**Comment by Dalkil** [comment]
> Can you guys try manually clearing the cache as ellesde pointed here [#43 (comment)](https://github.com/Zarroc2762/E7-Gear-Optimizer/issues/43#issuecomment-615913983) and see if works?
 Yes, Raphael, I did that once I could manage to get the archive loaded. I lost some heroes, but in the end I only had to insert them again, put their gear, and done. It's no longer a problem, just a small annoyance. 

Thanks for your reply.

**Comment by RaphaelDDL** [comment]
Odd that characters were lost but great it worked. Maybe add to ellesde and Zarroc's investigation list as to why.
Do you remember which heroes were lost?

**Comment by Dalkil** [comment]
> Odd that characters were lost but great it worked. Maybe add to ellesde and Zarroc's investigation list as to why.
> Do you remember which heroes were lost?

Yes, I remember four of the six that had gear. I don't know if there were other ones. But to be honest here, I think this was my fault. I've seen the characters you register have an id.number. I had **all** my characters in the list even though I had not geared them. That means I could have had more than the maximum amount allowed by the program. 

I say this because when I tried to add them, I got an error message which dissapeared as soon as I erased the characters I never use. After that, the problem was solved and I could add those six characters geared that I lost: Among them were Charles, Lilibet, Angelic Montmorancy, Cermia...

But as I told you, I think this is a problem on my end. 

By the way, could a space for Special Equipment be added? Is there a suggesions section over the site? I am not familiarized with github. Sorry if my question is not apporpriate.

**Comment by phootrf** [comment]
> 
> 
> Can you guys try manually clearing the cache as ellesde pointed here [#43 (comment)](https://github.com/Zarroc2762/E7-Gear-Optimizer/issues/43#issuecomment-615913983) and see if works?

I never used the cache function. To try it out I enabled it, cleared it, enabled it again. The error message still persists, however now it points to position 20983 (rest of the error message stays the same). I  checked what still got imported and it makes me lose 7 out of 19 registered hero data, so 12 got imported. However I retain all of my 142 registered gear data.

**Comment by greatmany** [comment]
I had the same problem too and managed to fix it by manually deleting Tamarinne from my json

Here is how the offending portion looked like in my json

      "ID": "3",
      "Name": "Tamarinne",
      "Gear": [
        {
          "ID": "zG"
        },
        {
          "ID": "zzzh"
        },
        {
          "ID": "zz]]"
        },
        {
          "ID": "zzz]]"
        },
        {
          "ID": "zzzu"
        },
        {
          "ID": "zzzK"
        }
      ],
      "Artifact": {
        "ATK": 65.0,
        "HP": 592.0
      },
      "Lvl": 60,
      "Awakening": 5,
      "Skills": [
        {
          "Enhance": 0
        },
        {
          "Enhance": 0
        },
        {
          "Enhance": 0
        }
      ]
    }



I've tried adding Tamarinne back as a new hero again after successfully importing but it gives the position 20987 error again whenever i try to do so

**Comment by phootrf** [comment]
> 
> 
> I had the same problem too and managed to fix it by manually deleting Tamarinne from my json
> 
> Here is how the offending portion looked like in my json
> 
> ```
>   "ID": "3",
>   "Name": "Tamarinne",
>   "Gear": [
>     {
>       "ID": "zG"
>     },
>     {
>       "ID": "zzzh"
>     },
>     {
>       "ID": "zz]]"
>     },
>     {
>       "ID": "zzz]]"
>     },
>     {
>       "ID": "zzzu"
>     },
>     {
>       "ID": "zzzK"
>     }
>   ],
>   "Artifact": {
>     "ATK": 65.0,
>     "HP": 592.0
>   },
>   "Lvl": 60,
>   "Awakening": 5,
>   "Skills": [
>     {
>       "Enhance": 0
>     },
>     {
>       "Enhance": 0
>     },
>     {
>       "Enhance": 0
>     }
>   ]
> }
> ```
> 
> I've tried adding Tamarinne back as a new hero again after successfully importing but it gives the position 20987 error again whenever i try to do so

How were you able to identify the culprit? I have no idea which data part in my json is causing the hiccups.

**Comment by Zarroc2762** [comment]
I managed to reproduce the issue on another system with a chinese language pack. Seems like the c# webclient did not download the JSON data in UTF-8 from epicsevendb. Therefore the star character in Tamarinne's skill produced the errors. This should be fixed in the latest release. Please test if it works for you. 

Some heroes disappeared in your cases because the optimizer imported everything up until the point of failure. So depending on where Tamarinne was located in your JSON file more or less heroes were imported before the optimizer reached Tamarinne and stopped. I added a new dailog to allow importing the rest of your heroes if one hero could not be imported in the future.

**Comment by phootrf** [comment]
I've been testing the latest release and I'm happy to report no import errors happened. All my units and gears could be imported, manipulated and saved. The saved database file under 2.8.5 could also be re read without issues.
